spec/models/course_spec.rb

```diff
@@ -0,0 +1,130 @@
+require "rails_helper"
+
+RSpec.describe Course, type: :model do
+  let(:course) {FactoryBot.build :course}
+  let(:invalid_course) {FactoryBot.build :course, name: nil}
+  let!(:course_two) do
+    FactoryBot.create :course, status: :active,
+                      name: "Luu Dai",
+                      created_at: "29-11-2020".to_date,
+                      updated_at: "29-11-2020".to_date
+  end
+  let!(:course_three) do
+    FactoryBot.create :course, status: :unactive,
+                      name: "Hoang Hai",
+                      description: "This is a sample description",
+                      created_at: "30-11-2020".to_date,
+                      updated_at: "30-11-2020".to_date
+  end
+  let!(:course_four) do
+    FactoryBot.create :course, status: :expired,
+                      name: "Tran Tri",
+                      description: "This is not a sample description",
+                      created_at: "01-12-2020".to_date,
+                      updated_at: "01-12-2020".to_date
+  end
+
+  describe "Validations" do
+    it "valid all fields" do
+      expect(course.valid?).to eq true
+    end
+
+    it "invalid any fields" do
+      expect(invalid_course.valid?).to eq false
+    end
+  end
+
+  describe "Associations" do
+    it "has many users" do
+      is_expected.to have_many :users
+    end
+
+    it "has many course_lecture" do
+      is_expected.to have_many :course_lecture
+    end
+  end
+
+  describe "Nested attributes" do
+    it "course lecture" do
+      is_expected.to accept_nested_attributes_for(:course_lecture)
+                 .allow_destroy true
+    end
+  end
+
+  describe "Enums" do
+    it "status" do
+      is_expected.to define_enum_for(:status)
+                 .with_values unactive: 0, active: 1, on_progress: 2, expired: 3
+    end
+  end
+
+  describe "Scope order by start date" do
```

Comment:
```suggestion
  describe ".order_by_created_at" do
```
đọc đến đoạn đặt tên cho class method chưa e (tat)

Reply:
Vâng ạ